### ISSUE TITLE
Base branch tests run even if current branch fails

### ISSUE THREAD

**Issue by dstoyanoff** [ArtiomTr/jest-coverage-report-action#222]
### Describe a bug
When the tests fail for the current branch, the base branch would still be executed

### Expected behavior
I see no point in executing the base branch as we cannot compare coverage anyway, so this is resulting in longer builds and more minutes consumed.

### Details

-   Action version: 2.0.2
-   OS, where your action is running (windows, linux): ubuntu-latest

**Comment by ArtiomTr** [comment]
Hello @dstoyanoff :wave:,

In theory, code coverage does not depend on the result of the test. The fact that a test fails does not mean that the behavior was not tested. Jest generates a report file even if all tests fail. So we can still compare the coverage of the two branches.

I don't know exactly your use case, but I think the main problem is that this action is slow.

I had a few ideas how to speed up the action, but I think it will take a long time to implement. One of them is to cache `base` branch report, as long as it does not change. This will drastically reduce action's execution time. However, this is not possible to do for now, because this action is not able to use `@actions/cache` nor `@actions/artifact` packages. I will describe problem more widely in #136 issue, if you're interested in it.

**tl;dr**

I would not like to cut some functionality, to speed up this action. There is some approaches, how to make this action faster, but this requires a lot of time and work.  I'm closing this issue, but if I've misunderstood you, feel free to reopen issue/ping me to discuss another solution!

**Comment by dstoyanoff** [comment]
Hey @ArtiomTr, got your points, thanks for your fast response.

Seems like my case was that the jest process crashed during test of the current branch due to a unhandled exception somewhere in the code.

May I suggest skipping the base branch tests if the jest doesn't exit with 0? That would mean that it crashed and most probably couldn't collect coverage results

**Comment by ArtiomTr** [comment]
@dstoyanoff,

There is a small issue. Jest always returns 1 as exit code, when it fails. It's not easy to determine, if it crashed, or failed in tests, or coverage threshold not passed. However, I think I could detect that by checking if `report.json` file exists. I'll try to implement this behavior, and see, if it works for your case. I'll create a separate issue for that problem.
